src/common/hashtable/src/hashjoin_hashtable.rs

```diff
@@ -108,10 +162,144 @@ where
     type Key = K;
 
     // Using hashes to probe hash table and converting them in-place to pointers for memory reuse.
-    fn probe(&self, hashes: &mut [u64]) {
-        hashes
-            .iter_mut()
-            .for_each(|hash| *hash = self.pointers[(*hash & self.hash_mask) as usize]);
+    fn probe(&self, hashes: &mut [u64], bitmap: Option<Bitmap>) -> usize {
+        let mut valids = None;
+        if let Some(bitmap) = bitmap {
+            if bitmap.unset_bits() == bitmap.len() {
+                hashes.iter_mut().for_each(|hash| {
+                    *hash = 0;
+                });
+                return 0;
+            } else if bitmap.unset_bits() > 0 {
+                valids = Some(bitmap);
+            }
+        }
```

Comment:
How about

```rust
if let Some(bitmap) = bitmap {
    if bitmap.unset_bits() == bitmap.len() {
        hashes.iter_mut().for_each(|hash| {
            *hash = 0;
        });
        return 0;
}
....
match bitmap {

}               
```

and we can extract a helper function for the code snippet
```rust
  let header = self.pointers[(*hash >> self.hash_shift) as usize];
  if header != 0 {
      *hash = remove_header_tag(header);
      count += 1;
  } else {
      *hash = 0;
  }
```

Reply:
> How about
> 
> ```rust
> if let Some(bitmap) = bitmap {
>     if bitmap.unset_bits() == bitmap.len() {
>         hashes.iter_mut().for_each(|hash| {
>             *hash = 0;
>         });
>         return 0;
> }
> ....
> match bitmap {
> 
> }               
> ```

The code in the PR will set bitmap to `None` when bitmap.unset_bits() == 0.